MoneyDataType/KnownCurrencyTable.cs

```diff
@@ -1,49 +1,53 @@
+using Money.Abstractions;
+using System;
 using System.Collections.Generic;
 using System.Globalization;
 using System.Linq;
-using Money.Abstractions;
 
-namespace Money
+namespace Money;
+
+internal static class KnownCurrencyTable
 {
-    internal static class KnownCurrencyTable
-    {
-        private static readonly object Obj = new object();
+    private static readonly object _obj = new();
 
-        internal static IDictionary<string, ICurrency> CurrencyTable { get; private set; }
+    internal static IDictionary<string, ICurrency> CurrencyTable { get; private set; }
 
-        internal static void EnsureCurrencyTable()
-        {
-            if (CurrencyTable == null)
-                InitCurrencyCodeTable();
-        }
+    internal static void EnsureCurrencyTable()
+    {
+        if (CurrencyTable == null) InitCurrencyCodeTable();
+    }
 
-        private class CurrencyEqualityComparer : IEqualityComparer<ICurrency>
-        {
-            public bool Equals(ICurrency x, ICurrency y) => x.CurrencyIsoCode == y.CurrencyIsoCode;
-            public int GetHashCode(ICurrency obj) => obj.CurrencyIsoCode.GetHashCode();
-        }
+    private sealed class CurrencyEqualityComparer : IEqualityComparer<ICurrency>
+    {
+        public bool Equals(ICurrency left, ICurrency right) =>
+            (left is null && right is null) ||
+            left?.CurrencyIsoCode?.Equals(right?.CurrencyIsoCode, StringComparison.OrdinalIgnoreCase) == true;
```

Comment:
This would be shorter and clearer:
```suggestion
            StringComparer.OrdinalIgnoreCase.Equals(left?.CurrencyIsoCode, right?.CurrencyIsoCode);
```